lib/blocks.php

```diff
@@ -147,6 +150,46 @@ function gutenberg_reregister_core_block_types() {
 
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
 
+/**
+ * Registers block styles for a core block.
+ *
+ * @param string $block_name The block-name.
+ *
+ * @return void
+ */
+function gutenberg_register_core_block_styles( $block_name ) {
+	if ( ! gutenberg_should_load_separate_block_styles() ) {
+		return;
+	}
+
+	$block_name = str_replace( 'core/', '', $block_name );
+
+	$style_path        = is_rtl()
+		? "build/block-library/blocks/$block_name/style-rtl.css"
+		: "build/block-library/blocks/$block_name/style.css";
+	$editor_style_path = is_rtl()
+		? "build/block-library/blocks/$block_name/style-editor-rtl.css"
+		: "build/block-library/blocks/$block_name/style-editor.css";
```

Comment:
@aristath You can use `wp_style_add_data( 'wp-block-' . $block_name, 'rtl', 'suffix' )` instead of such conditionals

Reply:
(or `wp_style_add_data( 'wp-block-' . $block_name . 'editor', 'rtl', 'suffix' )`, respectively)

Reply:
Does it mean you register both and WordPress decides which one to enqueue? That would be nice 👍🏻 

Do you have some recommendations how to add RTL handling for the block registration from `block.json` in WP Core. In this existing method:
https://github.com/WordPress/wordpress-develop/blob/3d43e57237e97c6421bda61774b91cf2b6fa2d49/src/wp-includes/blocks.php#L152-L171

Reply:
You register the style _once_, but which files are loaded depends on what you use.

**(A)** When using `wp_style_add_data( $handle, 'rtl', 'replace' )`:

_Instead_ of e.g. `awesome-styles.css`, WP will load `awesome-styles-rtl.css`.

Useful when using something like RTLCSS to automatically create a CSS file with flipped styles.

**(B)** When using `wp_style_add_data( $handle, 'rtl', 'url/to/my-additional-rtl-styles.css' )`:

_In addition to_ `awesome-styles.css`, WP will load `my-additional-rtl-styles.css`.

Useful when the additional file contains only the flipped styles, but everything else is in the main stylesheet.

See https://github.com/WordPress/wordpress-develop/blob/3d43e57237e97c6421bda61774b91cf2b6fa2d49/src/wp-includes/class.wp-styles.php#L253-L279

---

> Do you have some recommendations how to add RTL handling for the block registration from `block.json` in WP Core.

Good question! Not sure what's the best way at first glance. But would be good to open an issue fo discussion.

/cc @ocean90 in case he has ideas.

Reply:
Thank you @swissspidy! I made a PR in https://github.com/WordPress/gutenberg/pull/28274 with the changes suggested above :+1: